mapper/container_ports.go

```diff
@@ -20,26 +21,65 @@ func (n ContainerPorts) FromInternal(data map[string]interface{}) {
 	field.FromInternal(data)
 
 	containers := convert.ToInterfaceSlice(data["containers"])
-	ports := convert.ToInterfaceSlice(data["ports"])
+	annotationPorts := convert.ToInterfaceSlice(data["ports"])
+	annotationsPortsMap := map[string]map[string]interface{}{}
 
-	for i := 0; i < len(ports) && i < len(containers); i++ {
+	// process fields defined via annotations
+	for i := 0; i < len(annotationPorts) && i < len(containers); i++ {
 		container := convert.ToMapInterface(containers[i])
 		if container != nil {
-			portsSlice := convert.ToInterfaceSlice(ports[i])
-			var containerPorts []interface{}
+			portsSlice := convert.ToInterfaceSlice(annotationPorts[i])
+			portMap := map[string]interface{}{}
```

Comment:
Is it always that the "containers[i]" will be associated with annotationPorts[i] ?

Reply:
yes